src/rgw/rgw_sal_motr.cc

```diff
@@ -3633,6 +3634,11 @@ int MotrStore::next_query_by_name(string idx_name,
     else
       next_key = key_out[i + k - 1] + " ";
     ldout(cctx, 0) << "do_idx_next_op(): next_key=" << next_key << dendl;
+    // Resizing keys & vals vector when `keys_left < batch size`.
+    if(keys_left < (int)nr_kvp) {
+      keys.resize(keys_left);
+      vals.resize(keys_left);
+    }
```

Comment:
Vector resize is quite expensive operation. I'd recommend adding the check for index at the for loop at line 3605 above.

Reply:
let's say the remaining key is 1 and the batch size is 100, then first fetching 100 keys and then while copying values  to resultant vector adding a check will be more expensive than vector resizing.

Reply:
We can avoid resizing keys vector since we are considering only `vals.size()` in `do_idx_next_op()`, but logically it didn't feel good to me, so I didn't.